service/collector/collector.go

```diff
@@ -0,0 +1,67 @@
+package collector
+
+import (
+	"sync"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	Namespace = "chart_operator"
+)
+
+type Config struct {
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
+}
+
+type Collector struct {
+	g8sClient versioned.Interface
+	logger    micrologger.Logger
+}
+
+func New(config Config) (*Collector, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	c := &Collector{
+		g8sClient: config.G8sClient,
+		logger:    config.Logger,
+	}
+
+	return c, nil
+}
+
+func (c *Collector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- chartConfigDesc
+}
+
+func (c *Collector) Collect(ch chan<- prometheus.Metric) {
+	c.logger.Log("level", "debug", "message", "collecting metrics")
+
+	collectFuncs := []func(chan<- prometheus.Metric){
+		c.collectChartConfigStatus,
+	}
```

Comment:
Good to use the updated approach here with the WaitGroup, As we may need to add more collector funcs later.